Healex.HL7v2Anonymizer/Services/Anonymizer.cs

```diff
@@ -3,87 +3,88 @@
 using System.Linq;
 using static Healex.HL7v2Anonymizer.ReplacementOptions;
 
-namespace Healex.HL7v2Anonymizer.Services
-{
-    public class Anonymizer
-    {
-        private readonly ReplacementOptions _replacementOptions;
-
-        public Anonymizer(ReplacementOptions replacementOptions)
-        {
-            _replacementOptions = replacementOptions;
+namespace Healex.HL7v2Anonymizer.Services {
+
+    public class Anonymizer {
+
+        #region Private Members
+
+        private readonly ReplacementOptions options;
+
+        #endregion
+
+        #region Constructors
+
+        public Anonymizer(ReplacementOptions options) {
+            this.options = options;
         }
 
-        public bool Anonymize(Message message)
-        {
+        #endregion
+
+        #region Public API
+
+        public bool Anonymize(Message message) {
             var isSuccess = true;
+            for (var index = 0; index < message.SegmentCount; index++) {
+                var segment = message.Segments()[index]; 
+                var replacements = options.Segments.FirstOrDefault(s => s.Segment == segment.Name);
+                if (replacements != null) {
+                    message.Segments()[index] = new Segment("DummySegment", new HL7Encoding());
+                }
+
 
-            for (var segmentIndex = 0; segmentIndex < message.SegmentCount; segmentIndex++)
-            {
-                var segment = message.Segments().ElementAt(segmentIndex);
-                if (_replacementOptions.Segments.FirstOrDefault(segRep => segRep.Segment == segment.Name)
-                    is { } segmentReplacement)
-                {
+                if (options.Segments.FirstOrDefault(segRep => segRep.Segment == segment.Name) is { } segmentReplacement) {
                     // Create new temporary message for each repeating segment 
                     // because we can't set values in all repeating segments at once
-                    var tempMessage = AddSegmentAtIndex(segment, segmentIndex);
-                    foreach (Replacement replacement in segmentReplacement.Replacements)
-                    {
+                    var tempMessage = AddSegmentAtIndex(segment, index);
```

Comment:
Something is not right here. We're getting the applicable `SegmentReplacement` here 2times in every iteration of the for-loop. The first half does the same as `AddSegmentAtIndex`. So duplication?